crates/driver/src/domain/competition/score.rs

```diff
@@ -76,31 +89,102 @@ impl From<eth::U256> for ObjectiveValue {
     }
 }
 
-/// Comparing scores and objective values is needed to make sure the score is
-/// not higher than the objective value, which is a requirement for the score to
-/// be valid.
-impl std::cmp::PartialEq<ObjectiveValue> for Score {
-    fn eq(&self, other: &ObjectiveValue) -> bool {
-        self.0.eq(&other.0)
+impl std::ops::Add for ObjectiveValue {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self(self.0 + other.0)
     }
 }
```

Comment:
We implement `Add` for `ObjectiveValue + ObjectiveValue`. But if `self.0 == 0` or `other.0 == 0` then we can't really return an output that makes sense because of the saturating sub. I believe this can't happen right now, however it looks like it could be a source of issues eventually.

Reply:
Resolved with https://github.com/cowprotocol/services/pull/2022/commits/a7ecb3813946118799b1a932d636b042f15cb773